init.lua

```diff
@@ -1,29 +1,70 @@
-
 stamina = {}
+local modname = minetest.get_current_modname()
+local enable_damage = minetest.settings:get_bool("enable_damage")
+local armor_mod = minetest.get_modpath("3d_armor") and minetest.global_exists("armor") and armor.def
+local player_monoids_mod = minetest.get_modpath("player_monoids") and minetest.global_exists("player_monoids")
 
-STAMINA_TICK = 800		-- time in seconds after that 1 stamina point is taken
-STAMINA_TICK_MIN = 4		-- stamina ticks won't reduce stamina below this level
-STAMINA_HEALTH_TICK = 4		-- time in seconds after player gets healed/damaged
-STAMINA_MOVE_TICK = 0.5		-- time in seconds after the movement is checked
-
-STAMINA_EXHAUST_DIG = 3		-- exhaustion increased this value after digged node
-STAMINA_EXHAUST_PLACE = 1	-- .. after digging node
-STAMINA_EXHAUST_MOVE = 1.5	-- .. if player movement detected
-STAMINA_EXHAUST_JUMP = 5	-- .. if jumping
-STAMINA_EXHAUST_CRAFT = 20	-- .. if player crafts
-STAMINA_EXHAUST_PUNCH = 40	-- .. if player punches another player
-STAMINA_EXHAUST_LVL = 160	-- at what exhaustion player saturation gets lowered
+function stamina.log(level, message, ...)
+	return minetest.log(level, ("[%s] %s"):format(modname, message:format(...)))
+end
 
-STAMINA_HEAL = 1		-- number of HP player gets healed after STAMINA_HEALTH_TICK
-STAMINA_HEAL_LVL = 5		-- lower level of saturation needed to get healed
-STAMINA_STARVE = 1		-- number of HP player gets damaged by stamina after STAMINA_HEALTH_TICK
-STAMINA_STARVE_LVL = 3		-- level of staturation that causes starving
+if not enable_damage then
+	stamina.log("warning", "Stamina is disabled when damage is disabled.")
```

Comment:
```Lua
	return
end
```

Skips the entire code below.

Reply:
Yes it does. It seems like if you don't have damage enabled on your server, you'd just not install this mod, right? I'm not sure what the use case is, but the original code had something similar, though only for the event registrations at the end of the script. I suppose I should just get rid of this check entirely.

Reply:
Either way is fine. Just keep in mind that this mod might be enabled in worlds where singleplayer changes from creative+immortal to survival sometimes.

Reply:
Alright, I've added back the check for enable_damage, but in a cleaner way.